strongbox-cron/strongbox-cron-rest/src/main/java/org/carlspring/strongbox/cron/controller/CronTaskController.java

```diff
@@ -4,195 +4,272 @@
         {
-            logger.error(e.getMessage(), e);
-            return ResponseEntity.status(HttpStatus.BAD_REQUEST).body(e.getMessage());
+            return getExceptionResponseEntity(HttpStatus.INTERNAL_SERVER_ERROR, FAILED_SAVE_CONFIGURATION, e,
+                                              acceptHeader);
         }
     }
 
-    @ApiOperation(value = "Used to delete the configuration")
-    @ApiResponses(value = { @ApiResponse(code = 200, message = "The configuration was deleted successfully."),
-                            @ApiResponse(code = 400, message = "An error occurred.") })
-    @RequestMapping(value = "/cron", method = RequestMethod.DELETE)
-    public ResponseEntity deleteConfiguration(@RequestParam("name") String name)
+    @ApiOperation(value = "Used to update an existing configuration")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = SUCCESSFUL_UPDATE_CONFIGURATION),
+                            @ApiResponse(code = 400, message = FAILED_UPDATE_CONFIGURATION) })
+    @PutMapping(value = "{UUID}",
+                consumes = MediaType.APPLICATION_JSON_VALUE,
+                produces = { MediaType.TEXT_PLAIN_VALUE,
+                             MediaType.APPLICATION_JSON_VALUE })
+    public ResponseEntity updateConfiguration(@PathVariable("UUID") String uuid,
+                                              @RequestBody
+                                              @Validated(CronTaskConfigurationForm.ExistingConfiguration.class)
+                                              CronTaskConfigurationForm cronTaskConfigurationForm,
+                                              BindingResult bindingResult,
+                                              @RequestHeader(HttpHeaders.ACCEPT) String acceptHeader)
     {
-        final List<Exception> errors = new LinkedList<>();
-        final CronTaskConfigurationDto config = cronTaskConfigurationService.getTaskConfigurationDto(name);
-        if (config != null)
+        if (bindingResult.hasErrors())
         {
-            try
-            {
-                cronTaskConfigurationService.deleteConfiguration(config.getName());
-                if (config.contains("jobClass"))
-                {
-                    Class c = Class.forName(config.getProperty("jobClass"));
-                    Object classInstance = c.getConstructor().newInstance();
-
-                    logger.debug("> " + c.getSuperclass().getCanonicalName());
+            throw new RequestBodyValidationException(FAILED_UPDATE_CONFIGURATION, bindingResult);
+        }
 
-                    if (classInstance instanceof GroovyCronJob)
-                    {
-                        Path path = Paths.get(config.getProperty("script.path"));
-                        Files.deleteIfExists(path);
-                    }
-                }
-            }
-            catch (Exception e)
+        try
+        {
+            if (!StringUtils.equals(uuid, cronTaskConfigurationForm.getUuid()))
```

Comment:
The `UUID` should be taken from the submitted `URL` via `@Put(value = '{UUID}')` :) 
The form itself will not contain the `UUID` parameter (more on this in a comment below)